### PR TITLE
Rename methods and instance variables calling section uuid

### PR DESCRIPTION
**Pull request by chrislo**
In #1017 we want to store `Section`s in Mongo instead of having an in-memory representation. To make this refactoring easier we are renaming `Section#id` to `Section#uuid`. This PR completes the rename by using `uuid` consistently in method and variable names that previously referred to `id`.

The only places where I haven't renamed are:
- In old migrations
- In the parameters used in the `section` routes (which in turn are passed in the `params` hash to the Service objects). 

@floehopper - what do you think about this last omission? I think I could do something with `routes.rb` to rename this parameter, but I'm not sure it's worth the effort given that we may rename back to `id` from `uuid` once the conversion is complete.

### CODE REVIEW

File: app/exporters/section_publishing_api_exporter.rb [hunk 1]
```diff
@@ -13,17 +13,13 @@ def initialize(organisation, manual, section, update_type: nil)
...
   end
 
 private
 
   attr_reader :organisation, :manual, :section
 
-  def content_id
-    section.uuid
-  end
-
```
Comment (floehopper) [inline app/exporters/section_publishing_api_exporter.rb]: Minor: This seems a bit orthogonal to the purpose of the PR. Also I'm doing some significant refactoring in this area and so it might make life easier from a merging point of view if this commit was left out.

File: spec/exporters/publishing_api_publisher_spec.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@
       it "asks the publishing api to publish the section with the specific update_type" do
         subject.call
 
-        expect(publishing_api).to have_received(:publish).with(section_id, "republish")
+        expect(publishing_api).to have_received(:publish).with(section_uuid, "republish")
       end
     end
   end
```
Comment (floehopper) [inline spec/exporters/publishing_api_publisher_spec.rb]: Minor: I think it would make things clearer if you distinguish this commit from the other 2 by appending "in `PublishingAPIPublisher`" to the subject line. Ditto for the other three commits.

**Comment by chrislo** [comment]
Force pushing before merging

**Comment by chrislo** [comment]
I removed the commit you suggested and improved the commit message on the other three. Could you re-approve if you're happy @floehopper?
